src/logic/safe/store/actions/createTransaction.ts

```diff
@@ -201,10 +201,16 @@ export class TxSender {
     if (!isFinalization && canSignOffChain) {
       try {
         const signature = await this.onlyConfirm(hardwareWallet)
-        this.onComplete(signature, confirmCallback)
+
+        if (signature) {
```

Comment:
When WalletConnect rejects, it doesn't throw an error but instead just sends an empty signature.

Reply:
Have you tested this with Safe-Safe owners? I thought we kind of accepted that this was the case.